backend/amd64/selection.ml

```diff
@@ -192,14 +200,62 @@ class selector = object (self)
 
 inherit Selectgen.selector_generic as super
 
-method! is_immediate op n =
+method! is_immediate_int op n =
   match op with
-  | Iadd | Isub | Imul | Iand | Ior | Ixor | Icomp _ | Icheckbound ->
+  | Iintop (Iadd | Isub | Imul | Iand | Ior | Ixor | Icomp _ | Icheckbound) ->
       is_immediate n
   | _ ->
-      super#is_immediate op n
+      super#is_immediate_int op n
 
-method is_immediate_test _cmp n = is_immediate n
+method! is_immediate_float op f =
+  match op with
+  | Ifloatop (Iaddf | Isubf | Imulf | Idivf) -> f <> +0.0
+  | _ -> false
+
+method is_immediate_test_int _cmp n = is_immediate n
+
+method is_immediate_test_float cmp f =
+  match cmp with
+  | Ifloattest _ -> f <> +0.0
+  | _ -> false
+
+method! memory_operands_supported op chunk =
+  match op, chunk with
+  | Ifloatop (Iaddf | Isubf | Imulf |Idivf), Double -> true
+  | Ispecific Isqrtf, Double -> true
+  | Iintop _, (Word_int | Word_val) -> true
+  | Iintop _, _ ->
+    (* The value loaded from memory needs to be extended before use in Iintop  *)
+    false
+  | Ifloatop (Icompf _ ), Double -> true
+  | Ifloatop (Inegf | Iabsf), _ -> false
+  | Ifloatofint, (Word_int | Word_val) -> true
+  | Iintoffloat, Double -> true
+  | (Ifloatop _ | Ispecific _ | Ifloatofint | Iintoffloat), _ -> false
```

Comment:
These catch-all pattern (not. the one under `Ispecific`)
mean that we will not be warned when a new constructor
is added.